### PR TITLE
Update expression and anomaly detector documentation

### PR DESCRIPTION
**Pull request by kkondaka**
### Description
Data Prepper supports arithmetic operators but the documentation is not updated to reflect that.
Also, Random Cut forest in anomaly detector processor support a new option which is not reflected in the documentation.

### Issues Resolved
_List any issues this PR will resolve, e.g. Closes [...]._

### Version
_List the OpenSearch version to which this PR applies, e.g. 2.14, 2.12--2.14, or all._

### Frontend features
_If you're submitting documentation for an OpenSearch Dashboards feature, add a video that shows how a user will interact with the UI step by step. A voiceover is optional._ 

### Checklist
- [ ] By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license and subject to the [Developers Certificate of Origin](https://github.com/opensearch-project/OpenSearch/blob/main/CONTRIBUTING.md#developer-certificate-of-origin).
For more information on following Developer Certificate of Origin and signing off your commits, please check [here](https://github.com/opensearch-project/OpenSearch/blob/main/CONTRIBUTING.md#developer-certificate-of-origin).

### CODE REVIEW

File: _data-prepper/pipelines/expression-syntax.md [hunk 1]
```diff
@@ -30,6 +30,9 @@ The following table lists the supported operators. Operators are listed in order
 |----------------------|-------------------------------------------------------|---------------|
 | `()`                 | Priority expression                                   | Left to right |
 | `not`<br> `+`<br>  `-`| Unary logical NOT<br>Unary positive<br>Unary negative | Right to left |
+| `*`, `/`             | Multiplication and Division operators                 | Left to right |
```
Comment (dlvenable) [inline _data-prepper/pipelines/expression-syntax.md:33]:
```suggestion
| `*`, `/`             | Multiplication and division operators                 | Left to right |
```

Division is not a proper noun and need not be capitalized.

File: _data-prepper/pipelines/expression-syntax.md [hunk 2]
```diff
@@ -30,6 +30,9 @@ The following table lists the supported operators. Operators are listed in order
...
+| `+`, `-`             | Addition and Subtraction operators                    | Left to right |
```
Comment (dlvenable) [inline _data-prepper/pipelines/expression-syntax.md:34]:
```suggestion
| `+`, `-`             | Addition and subtraction operators                    | Left to right |
```

Substraction is not a proper noun and need not be capitalized.

File: _data-prepper/pipelines/expression-syntax.md [hunk 3]
```diff
@@ -170,6 +227,9 @@ White space is optional around relational operators, regex equality operators, e
 | `()`                 | Priority expression      | Yes                  | `/a==(/b==200)`<br>`/a in ({200})`                             | `/status in({200})`                   |
 | `in`, `not in`       | Set operators            | Yes                  | `/a in {200}`<br>`/a not in {400}`                             | `/a in{200, 202}`<br>`/a not in{400}` |
 | `<`, `<=`, `>`, `>=` | Relational operators     | No                   | `/status < 300`<br>`/status>=300`                              |                                       |
+| `+`                  | String concat operator   | No                   | `/status_code + /message + "suffix"`
```
Comment (dlvenable) [inline _data-prepper/pipelines/expression-syntax.md:230]:
```suggestion
| `+`                  | String concatenation operator   | No                   | `/status_code + /message + "suffix"`
```

File: _data-prepper/pipelines/expression-syntax.md [hunk 4]
```diff
@@ -170,6 +227,9 @@ White space is optional around relational operators, regex equality operators, e
...
+| `+`, `-`             | Arithmetic Add and Subtract operators | No      | `/status_code + length(/message) - 2`
```
Comment (dlvenable) [inline _data-prepper/pipelines/expression-syntax.md:231]:
```suggestion
| `+`, `-`             | Arithmetic addition and subtraction operators | No      | `/status_code + length(/message) - 2`
```

File: _data-prepper/pipelines/expression-syntax.md [hunk 5]
```diff
@@ -170,6 +227,9 @@ White space is optional around relational operators, regex equality operators, e
...
+| `*`, `/`             | Multiply and Divide operators | No              | `/status_code * length(/message) / 3`
```
Comment (dlvenable) [inline _data-prepper/pipelines/expression-syntax.md:232]:
```suggestion
| `*`, `/`             | Multiplication and division operators | No              | `/status_code * length(/message) / 3`
```

**Review by dlvenable — DISMISSED** [review]

File: _data-prepper/pipelines/configuration/processors/anomaly-detector.md [hunk 6]
```diff
@@ -53,6 +53,7 @@ You can configure `random_cut_forest` mode with the following options.
 | `sample_size` | `256` | 100--2500 | The sample size used in the ML algorithm. |
 | `time_decay` | `0.1` | 0--1.0 | The time decay value used in the ML algorithm. Used as the mathematical expression `timeDecay` divided by `SampleSize` in the ML algorithm. |
 | `type` | `metrics` | N/A | The type of data sent to the algorithm. |
+| `output_after` | 32 | N/A | Indicates the number of events to consume before outputting anamolies |
```
Comment (vagimeli) [inline _data-prepper/pipelines/configuration/processors/anomaly-detector.md:56]:
```suggestion
| `output_after` | 32 | N/A | Specifies the number of events to process before outputting any detected anomalies. |
```

File: _data-prepper/pipelines/expression-syntax.md [hunk 7]
```diff
@@ -91,9 +93,64 @@ not /status_code in {200, 202}
...
+Arithmetic expressions allow you to do few basic arithmetic operations. Arithmetic expressions may be combined with conditional expressions to make more complex conditional expressions. The available arithmetic operators are `+`, `-`, `*`, and `/` for doing addition, subtraction, multiplication and division respectively. The syntax for using the arithmetic operators is as follows:
+
+```
+<Any> + <Any>
+<Any> - <Any>
+<Any> * <Any>
+<Any> / <Any>
+```
+
+The following are some example arithmetic expressions:
```
Comment (vagimeli) [inline _data-prepper/pipelines/expression-syntax.md:107]:
```suggestion
The following are example arithmetic expressions: 
```

File: _data-prepper/pipelines/expression-syntax.md [hunk 8]
```diff
@@ -91,9 +93,64 @@ not /status_code in {200, 202}
...
+
+```
+/value + length(/message)
+/bytes / 1024
+/value1 - /value2
+/TimeInSeconds * 1000
+```
+{% include copy-curl.html %}
+
+The following are some example arithmetic expressions used in conditional expressions :
```
Comment (vagimeli) [inline _data-prepper/pipelines/expression-syntax.md:117]:
```suggestion
The following are example arithmetic expressions that can be used in conditional expressions: 
```

File: _data-prepper/pipelines/expression-syntax.md [hunk 9]
```diff
@@ -91,9 +93,64 @@ not /status_code in {200, 202}
 ```
 {% include copy-curl.html %}
 
+### Arithmetic expressions
+
+Arithmetic expressions allow you to do few basic arithmetic operations. Arithmetic expressions may be combined with conditional expressions to make more complex conditional expressions. The available arithmetic operators are `+`, `-`, `*`, and `/` for doing addition, subtraction, multiplication and division respectively. The syntax for using the arithmetic operators is as follows:
```
Comment (vagimeli) [inline _data-prepper/pipelines/expression-syntax.md:98]:
```suggestion
Arithmetic expressions enable basic mathematical operations like addition, subtraction, multiplication, and division. These expressions can be combined with conditional expressions to create more complex conditional statements. The available arithmetic operators are +, -, *, and /. The syntax for using the arithmetic operators is as follows:
```

File: _data-prepper/pipelines/expression-syntax.md [hunk 10]
```diff
@@ -91,9 +93,64 @@ not /status_code in {200, 202}
...
+
+### String concatenation expressions
+
+String concatenation expressions allow you to concatenate strings to generate new strings. The concatenated strings can be used in conditional expressions too. The syntax for using string concatenation is as follows:
+
+```
+<String Variable or String Literal> + <String Variable or String Literal>
+```
+
+The following are some example string concatenation expressions:
```
Comment (vagimeli) [inline _data-prepper/pipelines/expression-syntax.md:134]:
```suggestion
The following are example string concatenation expressions:
```

File: _data-prepper/pipelines/expression-syntax.md [hunk 11]
```diff
@@ -91,9 +93,64 @@ not /status_code in {200, 202}
...
+
+```
+/name + "suffix"
+"prefix" + /name
+"time of " + /timeInMs + " ms"
+```
+{% include copy-curl.html %}
+
+The following are some example string concatenation expressions used in conditional expressions :
```
Comment (vagimeli) [inline _data-prepper/pipelines/expression-syntax.md:143]:
```suggestion
The following are example string concatenation expressions that can be used in conditional expressions:
```

File: _data-prepper/pipelines/expression-syntax.md [hunk 12]
```diff
@@ -91,9 +93,64 @@ not /status_code in {200, 202}
...
+```
+/value + length(/message) > 200
+/bytes / 1024 < 10
+/value1 - /value2 != /value3 + /value4
+```
+{% include copy-curl.html %}
+
+### String concatenation expressions
+
+String concatenation expressions allow you to concatenate strings to generate new strings. The concatenated strings can be used in conditional expressions too. The syntax for using string concatenation is as follows:
```
Comment (vagimeli) [inline _data-prepper/pipelines/expression-syntax.md:128]:
```suggestion
String concatenation expressions enable you to combine strings to create new strings. These concatenated strings can also be used within conditional expressions. The syntax for using string concatenation is as follows:
```

File: _data-prepper/pipelines/expression-syntax.md [hunk 13]
```diff
@@ -91,9 +93,64 @@ not /status_code in {200, 202}
...
+```
+/service + ".com" == /url
+"www." + /service != /url
+```
+{% include copy-curl.html %}
+
 ### Reserved symbols
 
-Reserved symbols are symbols that are not currently used in the expression syntax but are reserved for possible future functionality or extensions. Reserved symbols include `^`, `*`, `/`, `%`, `+`, `-`, `xor`, `=`, `+=`, `-=`, `*=`, `/=`, `%=`, `++`, `--`, and `${<text>}`.
+Reserved symbols are symbols that are not currently used in the expression syntax but are reserved for possible future functionality or extensions. Reserved symbols include `^`, `%`, `xor`, `=`, `+=`, `-=`, `*=`, `/=`, `%=`, `++`, `--`, and `${<text>}`.
```
Comment (vagimeli) [inline _data-prepper/pipelines/expression-syntax.md:153]:
```suggestion
Certain symbols, such as ^, %, xor, =, +=, -=, *=, /=, %=, ++, --, and ${<text>}, are reserved for potential future functionality or extensions, even though they are not currently used in the expression syntax. Reserved symbols include `^`, `%`, `xor`, `=`, `+=`, `-=`, `*=`, `/=`, `%=`, `++`, `--`, and `${<text>}`.
```

**Review by vagimeli — APPROVED** [review]
Please see the writing edits for clarity and conciseness. Once these changes are approved, I'll move this PR into editorial and then merge it.

**Comment by vagimeli** [comment]
@dlvenable I've completed the doc review and accepted your edits and comments. I also edited text for readability and style. Please review and approve at your availability, and then I'll move this forward for publishing. Thank you.

**Comment by vagimeli** [comment]
> @dlvenable I've completed the doc review and accepted your edits and comments. I also edited text for readability and style. Please review and approve at your availability, and then I'll move this forward for publishing. Thank you.

@kkondaka @vagimeli Doc review is complete and ready for your approval. Thanks!

File: _data-prepper/pipelines/configuration/processors/anomaly-detector.md (hunk 6 above)
Comment (vagimeli) [inline _data-prepper/pipelines/configuration/processors/anomaly-detector.md:56]:
```suggestion
| `output_after` | 32 | N/A | Indicates the number of events to consume before outputting anomalies. |
```

**Comment by vagimeli** [comment]
> > @dlvenable I've completed the doc review and accepted your edits and comments. I also edited text for readability and style. Please review and approve at your availability, and then I'll move this forward for publishing. Thank you.
> 
> @kkondaka @vagimeli Doc review is complete and ready for your approval. Thanks!

@kkondaka Please provide a status update for this PR. It's awaiting your acceptance of my doc review changes and @dlvenable approval. Thank you.

**Comment by vagimeli** [comment]
> > > @dlvenable I've completed the doc review and accepted your edits and comments. I also edited text for readability and style. Please review and approve at your availability, and then I'll move this forward for publishing. Thank you.
> > 
> > 
> > @kkondaka @vagimeli Doc review is complete and ready for your approval. Thanks!
> 
> @kkondaka Please provide a status update for this PR. It's awaiting your acceptance of my doc review changes and @dlvenable approval. Thank you.

@kkondaka @dlvenable Please provide a status update for this PR. It's awaiting your acceptance of my doc review changes and @dlvenable approval. Thank you.

**Comment by Naarcha-AWS** [comment]
Reviewed this once again, looks like the original tech review by David was implemented. Dismissed his review and added Melissa's doc review. Merging now.

**Pull request merged**